keras/src/layers/preprocessing/resizing.py

```diff
@@ -93,6 +93,14 @@ def call(self, inputs):
             fill_mode=self.fill_mode,
             fill_value=self.fill_value,
         )
+        if resized.dtype == inputs.dtype:
+            return resized
+        if inputs.dtype.is_integer:
+            resized = self.backend.numpy.round(resized)
+        resized = self.backend.numpy.clip(
```

Comment:
This line is only for the int case and so should be indented, right?

Note that `inputs.dtype.is_integer` is a TF API. It will not work in general.

Reply:
> This line is only for the int case and so should be indented, right?
Right.

> Note that inputs.dtype.is_integer is a TF API. It will not work in general.
I know that.  As far as i understand TFDataLayer subclasses should be executed only in tf.data...
If no, replacing `dtype.is_integer` is simple: `backend.is_int_dtype`

But replacing `dtype.min`/`dtype.max` is much harder. In that lines we need saturating cast, but it is implemented on OPs level, not at backend one.

---
Should i replace `dtype.is_integer` with `backend.is_int_dtype`?

Should i repeat saturating cast logic here? Or maybe you can suggest other solution?

Reply:
> I know that. As far as i understand TFDataLayer subclasses should be executed only in tf.data...

This is not the case. A `TFDataLayer` is a regular layer, that can be used in tf.data even with a different backend. When using a different backend and outside of tf.data, it executes in the target backend.

> Should i replace dtype.is_integer with backend.is_int_dtype?

Yes.

> Should i repeat saturating cast logic here?

Why not use saturating cast directly?

Reply:
> Why not use saturating cast directly?

Because it uses regular `backend.numpy.clip` and `backend.cast`. See https://github.com/keras-team/keras/blob/master/keras/src/ops/core.py#L871

I think that will cause an issue when backend != "tensorflow" and layer is called inside `tf.data` pipeline. Am i wrong?

One workaround i see is to move `backend` to parameters of [this function](https://github.com/keras-team/keras/blob/master/keras/src/ops/core.py#L871) with fallback to default backend it is not passed. 

Then  we can import it and call with `self.backend...`

Reply:
Got it, so the issue is that `saturating_cast` is implement in terms of `keras.src.backend` ops and not in target backend ops. Add `backend=None` as an argument in `def _saturate_cast(x, dtype, backend=None):` seems fine since it isn't public and we do something similar in a few other places (e.g. when using `smart_resize` with a `TFDataLayer`).